snack-bar-api/src/core/interactor/port/client/create-client-use-cases.port.ts

```diff
@@ -0,0 +1,5 @@
+import { Client } from '@/core/domain/client/client.entity';
+
+export abstract class CreateClientUseCasesPort {
```

Comment:
Instead of implementing a port for every use case, why don't we implement a abstract class with generic types so that we would have only one port for entity use cases?

Reference:

- https://www.typescriptlang.org/docs/handbook/generics.html
- https://medium.com/swlh/generics-abstract-classes-in-typescript-4f149fd1404d
- https://github.com/fmo00/agama-rep-crud/blob/main/src/datasource/ports/datasource-diet.port.ts